api_v3/services/CategoryEntryService.php

```diff
@@ -28,89 +28,24 @@ public function initService($serviceId, $serviceName, $actionName)
-			
-		//validate user is entiteld to assign entry to this category 
-		if (kEntitlementUtils::getEntitlementEnforcement() && $category->getContributionPolicy() != ContributionPolicyType::ALL)
-		{
-			$categoryKuser = categoryKuserPeer::retrievePermittedKuserInCategory($categoryEntry->categoryId, kCurrentContext::getCurrentKsKuserId());
-			if(!$categoryKuser)
-			{
-				KalturaLog::err("User [" . kCurrentContext::getCurrentKsKuserId() . "] is not a member of the category [{$categoryEntry->categoryId}]");
-				throw new KalturaAPIException(KalturaErrors::CANNOT_ASSIGN_ENTRY_TO_CATEGORY);
-			}
-			if($categoryKuser->getPermissionLevel() == CategoryKuserPermissionLevel::MEMBER)
-			{
-				KalturaLog::err("User [" . kCurrentContext::getCurrentKsKuserId() . "] permission level [" . $categoryKuser->getPermissionLevel() . "] on category [{$categoryEntry->categoryId}] is not member [" . CategoryKuserPermissionLevel::MEMBER . "]");
-				throw new KalturaAPIException(KalturaErrors::CANNOT_ASSIGN_ENTRY_TO_CATEGORY);
-			}
-				
-			if(!$categoryKuser->hasPermission(PermissionName::CATEGORY_EDIT) && !$categoryKuser->hasPermission(PermissionName::CATEGORY_CONTRIBUTE) &&
-				!$entry->isEntitledKuserEdit(kCurrentContext::getCurrentKsKuserId()) &&
-				$entry->getCreatorKuserId() != kCurrentContext::getCurrentKsKuserId())
-				throw new KalturaAPIException(KalturaErrors::CANNOT_ASSIGN_ENTRY_TO_CATEGORY);				
-		}
-		
-		$categoryEntryExists = categoryEntryPeer::retrieveByCategoryIdAndEntryId($categoryEntry->categoryId, $categoryEntry->entryId);
-		if($categoryEntryExists && $categoryEntryExists->getStatus() == CategoryEntryStatus::ACTIVE)
-			throw new KalturaAPIException(KalturaErrors::CATEGORY_ENTRY_ALREADY_EXISTS);
-		
-		if(!$categoryEntryExists)
+
+		try
 		{
 			$dbCategoryEntry = new categoryEntry();
+			$entryId = $categoryEntry->entryId;
+			$categoryId = $categoryEntry->categoryId;
+			$dbCategoryEntry = $dbCategoryEntry->add($entryId, $categoryId);
+			$dbCategoryEntry->save();
 		}
-		else
-		{
-			$dbCategoryEntry = $categoryEntryExists;
-		}
-		
-		$categoryEntry->toInsertableObject($dbCategoryEntry);
-		
-		/* @var $dbCategoryEnry categoryEntry */
-		$dbCategoryEntry->setStatus(CategoryEntryStatus::ACTIVE);
-		
-		if (kEntitlementUtils::getEntitlementEnforcement() && $category->getModeration())
-		{
-			$categoryKuser = categoryKuserPeer::retrievePermittedKuserInCategory($categoryEntry->categoryId, kCurrentContext::getCurrentKsKuserId());
-			if(!$categoryKuser ||
-				($categoryKuser->getPermissionLevel() != CategoryKuserPermissionLevel::MANAGER && 
-				$categoryKuser->getPermissionLevel() != CategoryKuserPermissionLevel::MODERATOR))
-				$dbCategoryEntry->setStatus(CategoryEntryStatus::PENDING);
-		}
-		
-		if ($category->getModeration() && 
-		   (kEntitlementUtils::getCategoryModeration() || $this->getPartner()->getEnabledService(KalturaPermissionName::FEATURE_BLOCK_CATEGORY_MODERATION_SELF_APPROVE)))
+		catch (Exception $ex)
```

Comment:
No need to check the type of the Exception, in the catch just change the type from Exception to kCoreException,
All Exception that are of different type will be thrown up automatically.